shared/src/balancer/swap/fixed_point.rs

```diff
@@ -295,4 +306,18 @@ mod tests {
             Bfp::zero()
         );
     }
+
+    #[test]
+    fn to_big_decimal() {
+        assert_eq!(BigDecimal::from(Bfp::zero()), BigDecimal::zero());
+        assert_eq!(BigDecimal::from(Bfp::one()), BigDecimal::from(1));
+        assert_eq!(BigDecimal::from(Bfp::from(500)), BigDecimal::from(500));
+        assert_eq!(
+            BigDecimal::from(Bfp::from_wei(U256::MAX)),
+            BigDecimal::from_str(
+                "115792089237316195423570985008687907853269984665640564039457.584007913129639935"
+            )
+            .unwrap()
+        );
```

Comment:
I'd like to also see a test that explicitly consider the case `1 > value > 0` to make sure that no weird rounding is happening. The last tested value is technically enough, but I think it's nicer to make the test case more explicit. You could use something like:
```
        assert_eq!(
            "0.4".parse::<Bfp>().unwrap(),
            "0.4".parse::<BigDecimal>().unwrap()
        );
```